cafebabel/articles/translations/models.py

```diff
@@ -38,8 +37,19 @@ def clean(self):
         except Article.DoesNotExist:
             pass
 
+    @classmethod
+    def check_duplicate(cls, sender, document, **kwargs):
+        if document.id:
```

Comment:
Thinking again about it, in case of an update we have to make the check too like `Translation.objects(original_article=document.original_article, language=document.language).count() == 1` or something more clever…

Reply:
You mean if the language was changed during the update?

Reply:
Yes, proposition:

```python
    @classmethod
    def check_duplicate(cls, sender, document, **kwargs):
        first = Translation.objects(original_article=document.original_article, 
                                    language=document.language).first()
        if (document.id and document.language != first.language) or first:
            raise errors.NotUniqueError(
                'A translation refering to the same article with this '
                'language already exists.')
```

It requires a dedicated test too.